### DISCUSSION TITLE
Protect Data with iPhone application

### DISCUSSION THREAD

**Discussion by ToptenDev** [community/community#53874] [Programming Help] [answered]
### Select Topic Area

Question

### Select Feature Area

- [ ] Issues
- [ ] Projects

### Body

I have an iPhone application that talks to a web service over SSL. I would like to distribute a key with the app which the app can use to identify itself to the web service (An application key). All apps can use the same key, so I can distribute the same key with every app.
How can I distribute this in a way that protects it from discovery or use by anything other than the application?
I am using Core Data with a SQLite db.

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by Krieit** [comment, marked as answer]
You can use the iOS keychain mechanism. Google for that for more info, or get a look at [SFHFKeychainUtils](https://github.com/ldandersen/scifihifi-iphone/tree/master/security), which is a small class by Buzz Andersen that makes handling the key chain very easy.
